### PR TITLE
Re-add support for dispatch/divide by days

### PR DESCRIPTION
**Pull request by maukoquiroga**
Fixes govzeroaotearoa/openfisca-aotearoa#15
Depends on #1141 

#### New features

- Introduce support for the ``day`` date unit in `holders.set_input_dispatch_by_period` and `holders.set_input_divide_by_period`
  - Allows for dispatching values per day, for example, to provide a daily (week, fortnight) to an yearly variable.
  - Inversely, allows for calculating the daily (week, fortnight) value of a yearly input.

### CODE REVIEW

**Comment by MattiSG** [comment]
I will postpone reviewing this PR until the one it depends on is approved, and suggest that this one is set to draft. Indeed, I fail to see how it could be approved when #1141 might not go through.

Do you truly need #1141 for this one to be reviewed? 🙂 As an alternative to switching to draft, I suggest to remove the dependency and rebase straight on `master`. Yes, `test-docs` might fail until #1141 is fixed, but that is true of all PRs, not only of this one. This would enable reviews to be parallelised and proper solutions found for #1140 without blocking all the others.

**Comment by maukoquiroga** [comment]
Done!

**Comment by maukoquiroga** [comment]
I'm not sure if this is a new feature, or a bug-fix as the behaviour this adds was introduced as a replacement of a behaviour that was deprecated, however the pull request forgot to include `days` in that migration.

**Comment by benjello** [comment]
@maukoquiroga : looks very good to me.
Just a question: how does this PR handles month day size, february and leeap year for divide_by_period ?
It does not seem to be tested although there are a bunch of tests for that in `Period` and `Instant`.

**Comment by maukoquiroga** [comment]
> @maukoquiroga : looks very good to me. Just a question: how does this PR handles month day size, february and leeap year for divide_by_period ? It does not seem to be tested although there are a bunch of tests for that in `Period` and `Instant`.

I didn't do a lot more than «reenabling» a feature that already existed.

If you see in the tests, month days are handled appropriately (31 + 28 + 30).

Maybe test should be more precise about leap years.

**Comment by maukoquiroga** [comment]
(By the way, I think that holders implements it's own logic in terms of equivalences between days, months and years. We should see which is better, and keep just one in periods, as holder should not handle that kind of logic IMHO).

**Comment by maukoquiroga** [comment]
@benjello @MattiSG could you please take a look at this pull request again?

**Comment by maukoquiroga** [comment]
cc @MattiSG @benjello

File: openfisca_core/holders/tests/test_helpers.py
Comment (benjello) [inline openfisca_core/holders/tests/test_helpers.py:42]: Why it is not 1095 = 3 * 365 ? (Is it because a year is approx. 395.25 ?) and you rounded the number ?
It is different from the division counter-part.

Comment (maukoquiroga) [reply openfisca_core/holders/tests/test_helpers.py:42]: It is just because there's a leap year with 366 days.

I'm counting calendar days instead of multiplying by 365.

**Review by benjello — APPROVED** [review]

**Pull request merged**
